File: Zisimopoulos_Valencia/Software Construction/QLS/src/uva/sc/qls/logic/StyleSheet.java
package uva.sc.qls.logic;

import java.util.List;
import java.util.ArrayList;

import uva.sc.qls.ast.IQLSNode;
import uva.sc.qls.ast.IQLSNodeVisitor;

public class StyleSheet implements IQLSNode {

	ID			id;
	List<Page>	pages	= new ArrayList<Page>();

	public StyleSheet(ID id, List<Page> pages) {
		this.id = id;
		this.pages = pages;
	}

	public ID getId() {
		return id;
	}

	public List<Page> getPages() {
		return pages;
	}

	public <T> T accept(IQLSNodeVisitor<T> visitor) {
		return visitor.visit(this);
	}

	public String toString() {
		String result = "[Stylesheet]: " + id.toString() + " {\n\t";
		for (Page page : pages) {
			result += page.toString();
		}
		result += "\n}";
		return result;
	}

}

File: geert.kai/lib/qls/ast/stylesheet.rb
module QLS
  module AST
    class StyleGroup
      attr_reader :name, :rules

      def initialize(name, rules)
        @name = name
        @rules = rules
      end

      def accept(visitor)
        visitor.visit(self)
      end
    end

    class Stylesheet < StyleGroup
    end

    class Page < StyleGroup
    end

    class Section < StyleGroup
    end
  end
end
